src/Serialization/Rings.jl

```diff
@@ -0,0 +1,112 @@
+################################################################################
+#  non simpleton base rings
+
+function load_internal(s::DeserializerState, t::Type{Nemo.NmodRing}, dict::Dict)
+    modulus = load_type_dispatch(s, UInt64, dict[:modulus])
+    
+    return t(modulus)
+end
+
+function save_internal(s::SerializerState, R::Nemo.NmodRing)
+    return Dict(
+        :modulus => save_type_dispatch(s, modulus(R))
+    )
+end
+
+################################################################################
+#  Polynomial Rings
+function save_internal(s::SerializerState, R::Union{MPolyRing, PolyRing})
+    return Dict(
+        :symbols => save_type_dispatch(s, symbols(R)),
+        :base_ring => save_type_dispatch(s, base_ring(R)),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{MPolyRing, PolyRing}},
+                       dict::Dict)
+    base_ring = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
+    symbols = load_type_dispatch(s, Vector{Symbol}, dict[:symbols]) 
+
+    if length(symbols) == 1
```

Comment:
if I understand right, then this code will serialize a `PolyRing` and an `MPolyRing` in one variable identically, and deserialize them as a `PolyRing`.

But you do not similarly merge `PolyElem` and `MPolyElem`. Thus I predict trying to serialize and then deserialize a `MPolyRing` with one variable (as created by e.g. `PolynomialRing(QQ,["x"])`) will not work correctly.

To fix this, I would recommend to distinguish between the two kinds of rings when serializing. This should be a simple matter of giving the second argument to this function a name, say `T`, and then checking this check to
```suggestion
    if T isa PolyRing
        @assert length(symbols) == 1
```

Reply:
I added the type check instead of number of symbols check, but I dont think the assert is necessary since it will throw an error if number of symbols is greater than 1 anyway